hack/tests/scaffolding/e2e-go-scaffold.sh

```diff
@@ -34,5 +52,8 @@ set -- "${ORIG_ARGS[@]}"
 go build -o $BASEPROJECTDIR/scaffold-memcached $ROOTDIR/hack/tests/scaffolding/scaffold-memcached.go
 
 pushd "$BASEPROJECTDIR"
+# scaffold-memcached uses "operator-sdk-old new ..." to scaffold the legacy project
+# since "new" is not present in the new CLI
+download_old_sdk_binary
```

Comment:
Nit: Does this download `operator-sdk-old` into the root of the operator-sdk project during the e2e test? If so, will it pollute a developer's project by leaving that file laying around after the e2e test completes?

If my understanding is correct, this is still non-blocking and could be done in a follow-up.

Reply:
@joelanford No, this is done right after `pushd "$BASEPROJECTDIR"` which is a tmp dir for the operator project directory.
So the binary will be pulled to the test project directory and not the SDK repo.